students/humberto_gonzalez/lesson05/assignment/database.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+"""
+Spyder Editor
+
+This is a temporary script file.
+"""
+# pylint: disable=too-many-locals,logging-format-interpolation,bare-except,line-too-long
+
+import logging
+import os
+import csv
+from pymongo import MongoClient
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+def import_data(directory_name, product_file, customer_file, rentals_file):
+    """imports data from csv file, returns counts of entries and errors"""
+
+    mongo = MongoDBConnection()
+
+    with mongo:
+        LOGGER.info("Establishing MongoDB connection")
+        database = mongo.connection.storeDB
+
+        LOGGER.info("Establishing databases")
+        customers = database["customers"]
+        products = database["products"]
+        rentals = database["rentals"]
+
+        #entry counts
+        customer_count = 0
+        product_count = 0
+        rental_count = 0
+
+        #error_counts
+        customer_errors = 0
+        product_errors = 0
+        rental_errors = 0
+
+        with open(os.path.join(directory_name, product_file)) as csv_file:
+
+            product_data = csv.reader(csv_file, delimiter=",")
+            for entry in product_data:
+                try:
+                    product_entry = {"product_id":entry[0],
+                                     "description":entry[1],
+                                     "product_type":entry[2],
+                                     "quantity_available":entry[3]}
+                    products.insert_one(product_entry)
+                    product_count += 1
+                    LOGGER.info(f"Added {entry[0]} to product database")
+                except:
```

Comment:
Avoid using bare or general exceptions in your code. They're ok for debugging, when you just need to force your code to run, but they present a significant risk for code intended for release (which is what these assignments try to reproduce), as they will capture _anything_ triggering an exception, including issues not seen during development and which might need different handling.